volatility3/framework/plugins/windows/modules.py

```diff
@@ -50,49 +59,74 @@ def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]
+            if file_handle:
+                file_output = file_handle.preferred_filename
+
+        return file_output
+
+    def process_module(self, session_layers, pe_table_name, mod):
+        if self.config["base"] and self.config["base"] != mod.DllBase:
+            return None
+
+        try:
+            BaseDllName = mod.BaseDllName.get_string()
+        except exceptions.InvalidAddressException:
+            BaseDllName = ""
+
+        if self.config["name"] and self.config["name"] not in BaseDllName:
+            return None
+
+        try:
+            FullDllName = mod.FullDllName.get_string()
+        except exceptions.InvalidAddressException:
+            FullDllName = ""
+
+        file_output = "Disabled"
+        if self.config["dump"]:
+            file_output = self.dump_module(session_layers, pe_table_name, mod)
+
+        return (
+            format_hints.Hex(mod.vol.offset),
+            format_hints.Hex(mod.DllBase),
+            format_hints.Hex(mod.SizeOfImage),
+            BaseDllName,
+            FullDllName,
+            file_output,
+        )
+
     def _generator(self):
         kernel = self.context.modules[self.config["kernel"]]
+
         pe_table_name = intermed.IntermediateSymbolTable.create(
             self.context, self.config_path, "windows", "pe", class_types=pe.class_types
         )
 
-        for mod in self.list_modules(
+        session_layers = list(
+            self.get_session_layers(
+                self.context, kernel.layer_name, kernel.symbol_table_name
+            )
+        )
+
+        for mod in self._enumeration_method(
             self.context, kernel.layer_name, kernel.symbol_table_name
         ):
-            try:
-                BaseDllName = mod.BaseDllName.get_string()
-            except exceptions.InvalidAddressException:
-                BaseDllName = ""
-
-            try:
-                FullDllName = mod.FullDllName.get_string()
-            except exceptions.InvalidAddressException:
-                FullDllName = ""
-
-            if self.config["name"] and self.config["name"] not in BaseDllName:
-                continue
-
-            file_output = "Disabled"
-            if self.config["dump"]:
-                file_handle = dlllist.DllList.dump_pe(
-                    self.context, pe_table_name, mod, self.open
-                )
-                file_output = "Error outputting file"
-                if file_handle:
-                    file_handle.close()
-                    file_output = file_handle.preferred_filename
-
-            yield (
-                0,
-                (
-                    format_hints.Hex(mod.vol.offset),
-                    format_hints.Hex(mod.DllBase),
-                    format_hints.Hex(mod.SizeOfImage),
-                    BaseDllName,
-                    FullDllName,
-                    file_output,
-                ),
-            )
+            record = self.process_module(session_layers, pe_table_name, mod)
```

Comment:
So this could be expanded to call `process_module` and then check the config here as to whether to dump or not.